### ISSUE TITLE
maxinsns flag does not work when ncpus > 1

### ISSUE THREAD

**Issue by kabylkas** [chipsalliance/dromajo#73]
Dromajo does not respond to maxinsns variable when ncpus > 1

To reproduce:

```
./dromajo --trace 0 --maxinsns 100000 --ncpus 2 [riscv-bin]
```

**Comment by kabylkas** [comment]
It seems the issue is related to the unsignedness of `maxinsns` or the fact that it is common to all cores. Once `maxinsns` riches zero, the next core in the loop overflows it here: 
https://github.com/chipsalliance/dromajo/blob/6f68f74a2a3ca26c5460f7864d5769face2c69bd/src/dromajo.cpp#L125

and this condition fails:
https://github.com/chipsalliance/dromajo/blob/6f68f74a2a3ca26c5460f7864d5769face2c69bd/src/dromajo.cpp#L127

**Comment by kabylkas** [comment]
Fixed in #76
